core/src/main/java/io/questdb/WorkerPoolManager.java

```diff
@@ -34,55 +34,46 @@
-    protected final WorkerPool sharedPool;
+    protected final WorkerPool sharedPoolNetwork;
+    // When parallel querying is disabled, query pool will be null. All Network and Write pools will always be created.
+    @Nullable
+    protected final WorkerPool sharedPoolQuery;
+    protected final WorkerPool sharedPoolWrite;
     private final AtomicBoolean closed = new AtomicBoolean();
     private final CharSequenceObjHashMap<WorkerPool> dedicatedPools = new CharSequenceObjHashMap<>(4);
     private final AtomicBoolean running = new AtomicBoolean();
 
     public WorkerPoolManager(ServerConfiguration config) {
-        sharedPool = new WorkerPool(config.getWorkerPoolConfiguration());
-        configureSharedPool(sharedPool); // abstract method giving callers the chance to assign jobs
+        sharedPoolNetwork = new WorkerPool(config.getNetworkWorkerPoolConfiguration());
+        sharedPoolQuery = config.getQueryWorkerPoolConfiguration().getWorkerCount() > 0 ? new WorkerPool(config.getQueryWorkerPoolConfiguration()) : null;
+        sharedPoolWrite = new WorkerPool(config.getWriteWorkerPoolConfiguration());
+
+        WorkerPool queryPool = sharedPoolQuery != null ? sharedPoolQuery : sharedPoolNetwork;
+        configureSharedPool(sharedPoolNetwork, queryPool, sharedPoolWrite); // abstract method giving callers the chance to assign jobs
         config.getMetrics().addScrapable(this);
     }
 
-    public WorkerPool getInstance(@NotNull WorkerPoolConfiguration config, @NotNull Requester requester) {
-        if (running.get() || closed.get()) {
-            throw new IllegalStateException("can only get instance before start");
-        }
-
-        if (config.getWorkerCount() < 1) {
-            LOG.info().$("using SHARED pool [requester=").$(requester)
-                    .$(", workers=").$(sharedPool.getWorkerCount())
-                    .I$();
-            return sharedPool;
-        }
+    public WorkerPool getInstanceNetwork(@NotNull WorkerPoolConfiguration config, @NotNull Requester requester) {
+        return getWorkerPool(config, requester, sharedPoolNetwork);
+    }
 
-        String poolName = config.getPoolName();
-        WorkerPool pool = dedicatedPools.get(poolName);
-        if (pool == null) {
-            pool = new WorkerPool(config);
-            dedicatedPools.put(poolName, pool);
-        }
-        LOG.info().$("new DEDICATED pool [name=").$(poolName)
-                .$(", requester=").$(requester)
-                .$(", workers=").$(pool.getWorkerCount())
-                .$(", priority=").$(config.workerPoolPriority())
-                .I$();
-        return pool;
+    public WorkerPool getInstanceWrite(@NotNull WorkerPoolConfiguration config, @NotNull Requester requester) {
+        return getWorkerPool(config, requester, sharedPoolWrite);
     }
 
-    public WorkerPool getSharedPool() {
-        return sharedPool;
+    public WorkerPool getSharedPoolNetwork() {
```

Comment:
these methods should be renamed as we lost concept of "shared" pool

Reply:
All three are still "shared" pools, as the opposite to "dedicated" pools.
Dedicated pools are still possible, you can turn them on for Pg, ILP, Wal Apply etc.